test/bin/opam-provided.t/run.t

```diff
@@ -0,0 +1,91 @@
+We have a package that should not be installed via opam. There might be various
+reasons we might not want that (e.g. it cannot be built with `dune`), it is
+meant as a helper binary etc.
+
+We start with a minimal opam-repository:
+
+  $ gen-minimal-repo
+
+We have a file that depends on package `b`.
+
+  $ opam show --no-lint --raw -fdepends ./vendored.opam
+  "dune" "b"
+
+This package does not set any `opam-provided` packages:
+
+  $ opam show --no-lint --raw -fx-opam-monorepo-opam-provided ./vendored.opam
+  
+
+It should lock successfully.
+
+  $ opam-monorepo lock vendored > /dev/null
+
+The lockfile should thus contain the package `b` and mark it as `vendor` since
+opam-monorepo will vendor it.
+
+  $ opam show --no-lint --raw -fdepends ./vendored.opam.locked | grep "\"b\""
```

Comment:
I think you can grep the file directly here, although I understand this is making it clear it comes from the dependencies of the lockfile, I'm wondering whether it's really necessary.

Reply:
It made me realize that I wrote a bunch of `cat file | grep` in the tests... I'll fix those!